src/pages/talks/[talk].astro

```diff
@@ -0,0 +1,195 @@
+---
+import { FavoriteButton } from "../../components/button/favoriteButton/FavoriteButton";
+import TalkLayout from "../../layouts/Layout.astro";
+import {
+  getDisciplineColorPair,
+  getDisciplineName,
+} from "../../utils/discipline";
+import { schedule, type Talk } from "../../utils/schedule";
+
+const exampleDescription =
+  "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.";
+const exampleSpeakerDescription =
+  "Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo";
+
+function getTalkById(talkId: string): Talk | undefined {
+  const allTalks: Talk[] = schedule
+    .flatMap((block) => block.tracks.flat())
+    .filter((scheduleEntry) => scheduleEntry.type === "talk");
+
+  return allTalks.find((t) => t.id === talkId);
+}
+
+const { talk: talkId } = Astro.params;
+
+if (talkId === undefined) {
+  return Astro.redirect("/404");
+}
+
+const talk = getTalkById(talkId);
+
+if (talk === undefined) {
+  return Astro.redirect("/404");
+}
+
+const colorPair = getDisciplineColorPair(talk.discipline);
+
+const disciplineBackgroundColor = colorPair.color;
+const disciplineTextColor = colorPair.textColor;
+---
+
+<TalkLayout>
+  <a class="talk__schedule_link" href="/schedule">
+    <span class="talk__left_arrow">↓</span>
+    <span class="talk__talk__link_text">Full schedule</span>
+  </a>
+  <main>
+    <section>
+      <h1>
+        {talk.title}
+      </h1>
+      <div class="talk__talk_section">
+        <div class="talk__talk_details">
+          <p class="talk__chip" id="talk__time_and_location">
+            {`${talk.location} ${talk.from}-${talk.to}`}
+          </p>
+          <p class="talk__chip" id="talk__discipline">
```

Comment:
Kan også fjerne id hvis den kun brukes for styling. kan ha flere klasser i class. feks class="talk_chip talk_discipline"